src/gameplay.js

```diff
@@ -36,31 +39,31 @@ const Gameplay = () => {
     }
 
 
-    const writeLeaderboard = () =>
-    {
+    const writeLeaderboard = () => {
         var playerToAdd = {
-            "name" : settings.playerName,
+            "name": settings.playerName,
             "score": score
         }
         return axios.post('http://localhost:5000/leaderboard', playerToAdd)
-        .then(function (response) {
-          console.log(response);
-          return response;
-        })
-        .catch(function (error) {
-          console.log(error);
-          return false;
-        });
+            .then(function (response) {
+                console.log(response);
+                return response;
+            })
+            .catch(function (error) {
+                console.log(error);
+                return false;
+            });
     }
-    
+
     const endGame = () => {
         setSpeed(null);
         setGameOver(true);
         writeLeaderboard();
     }
     const moveSnake = ({keyCode}) => {
-        if (keyCode < 37 || keyCode > 40)
+        if (gameOver || keyCode < 37 || keyCode > 40)
```

Comment:
Thank you for adding these logics to fix the gameplay bug.

Reply:
Of course! I was very determined as I wanted the gameplay to be as enjoyable as possible, and I knew how enjoyable snake can be. Line 66 was the most important component of the bugfix.